typing/untypeast.ml

```diff
@@ -330,7 +330,7 @@ let pattern : type k . _ -> k T.general_pattern -> _ = fun sub pat ->
     | Tpat_tuple list ->
         Ppat_tuple (List.map (sub.pat sub) list)
     | Tpat_construct (lid, _, args) ->
```

Comment:
I think that we should add something to the `Typedtree` to hold the names of the existientials and the typed type expression from the constraint. Then we should use that here rather than just assuming it is `None`.

Reply:
Done